src/objects/zcl_abapgit_object_bdef.clas.abap

```diff
@@ -0,0 +1,457 @@
+          <lv_objtype_tr> = ms_item-obj_type.
+          ASSIGN COMPONENT 'SUBTYPE_WB' OF STRUCTURE <ls_object_type_tmp> TO <lv_subtype_wb>.
+          <lv_subtype_wb> = 'BDO'.
+
+          " Remapping to API data structure due to backward compatibility to NW 7.02
+          CREATE DATA lr_object_name_tmp TYPE ('SEU_OBJKEY').
+          ASSIGN lr_object_name_tmp->* TO <lv_object_name_tmp>.
+          <lv_object_name_tmp> = ms_item-obj_name.
+
+          CALL METHOD ('zcl_WB_OBJECT_OPERATOR_FACTORY')=>('CREATE_OBJECT_OPERATOR')
+            EXPORTING
+              object_type = <ls_object_type_tmp>
+              object_key  = <lv_object_name_tmp>
+            RECEIVING
+              result      = lo_bdef.
+
+          CREATE DATA lr_data TYPE ('zcl_DDLX_WB_OBJECT_DATA=>TY_OBJECT_DATA').
+          ASSIGN lr_data->* TO <ls_data>.
+
+          CALL METHOD lo_bdef->('zif_WB_OBJECT_OPERATOR~READ')
+            EXPORTING
+              version        = 'A'
+              data_selection = 'AL'
+            IMPORTING
+              data           = <ls_data>.
+
+          ASSIGN COMPONENT 'METADATA' OF STRUCTURE <ls_data> TO <lg_struct>.
+          ASSIGN COMPONENT 'CHANGED_BY' OF STRUCTURE <lg_struct> TO <lg_field>.
+          if sy-subrc = 0.
+          rv_user = <lg_field>.
+          endif.
+
+        CATCH cx_root INTO lx_root.
+          zcx_abapgit_exception=>raise('BDEF read error').
+      ENDTRY.
+    ENDMETHOD.
+
+
+    METHOD zif_abapgit_object~compare_to_remote_version.
+      CREATE OBJECT ro_comparison_result TYPE zcl_abapgit_comparison_null.
+    ENDMETHOD.
+
+
+    METHOD zif_abapgit_object~delete.
+      DATA lx_root TYPE REF TO cx_root.
+
+      DATA:
+        lo_bdef            TYPE REF TO object,
+        lr_data            TYPE REF TO data,
+        lr_object_type_tmp TYPE REF TO data,
+        lr_object_name_tmp TYPE REF TO data.
+
+      FIELD-SYMBOLS:
+        <ls_data>            TYPE any,
+        <ls_object_type_tmp> TYPE any,
+        <lv_objtype_tr>      TYPE any,
+        <lv_subtype_wb>      TYPE any,
+        <lv_object_name_tmp> TYPE any.
+
+      TRY.
+          " Remapping to API data structure due to backward compatibility to NW 7.02
+          CREATE DATA lr_object_type_tmp TYPE ('WBOBJTYPE').
+          ASSIGN lr_object_type_tmp->* TO <ls_object_type_tmp>.
+
+          ASSIGN COMPONENT 'OBJTYPE_TR' OF STRUCTURE <ls_object_type_tmp> TO <lv_objtype_tr>.
+          <lv_objtype_tr> = ms_item-obj_type.
+          ASSIGN COMPONENT 'SUBTYPE_WB' OF STRUCTURE <ls_object_type_tmp> TO <lv_subtype_wb>.
+          <lv_subtype_wb> = 'BDO'.
+
+          " Remapping to API data structure due to backward compatibility to NW 7.02
+          CREATE DATA lr_object_name_tmp TYPE ('SEU_OBJKEY').
+          ASSIGN lr_object_name_tmp->* TO <lv_object_name_tmp>.
+          <lv_object_name_tmp> = ms_item-obj_name.
+
+          CALL METHOD ('zcl_WB_OBJECT_OPERATOR_FACTORY')=>('CREATE_OBJECT_OPERATOR')
```

Comment:
looks like the call to
`cl_WB_OBJECT_OPERATOR_FACTORY')=>('CREATE_OBJECT_OPERATOR'`
is identical in 5 places, can this be moved to private method?